src/main/ruby/core/truffle/cext.rb

```diff
@@ -1349,6 +1349,7 @@ def rb_yield_splat(values)
   end
 
   def rb_ivar_lookup(object, name, default_value)
+    name = "@#{name}" unless name.to_s.start_with?('@')
```

Comment:
Which test expects that? It sounds like a weird change.

Reply:
`rb_attr_get` is just `rb_ivar_lookup`

https://github.com/ruby/ruby/blob/c08f7b80889b531865e74bc5f573df8fa27f2088/variable.c#L1226

`openssl` calls `rb_attr_get(self, id_key_set)`, where `id_key_set = rb_intern_const("key_set")`

https://github.com/ruby/ruby/blob/4a375f3ef518797bc8788abf0891655d79aa719c/ext/openssl/ossl_cipher.c#L381
https://github.com/ruby/ruby/blob/4a375f3ef518797bc8788abf0891655d79aa719c/ext/openssl/ossl_cipher.c#L1061

But something seems to be wrong as I still get some failures that are close to this code.

Reply:
`rb_attr_get`, `rb_ivar_get` and `rb_iv_get` can all be called with names not starting with `@`.

Reply:
Which are meant to instance internal ivars not accessible from Ruby.

Reply:
Yes... that's what I've done isn't it?

Reply:
No, a `@` is never added in MRI, and both `foo` and `@foo` can be stored on the same object.

Reply:
Ok, well that wasn't working either then.

Reply:
I think we just did not have the case before.
The `object_ivar_defined?`, `object_ivar_get`, `object_ivar_set` primitives can be used to avoid the check on the name.